### PR TITLE
Updates for Python 3.8 compatibility

### PR DESCRIPTION
**Pull request by sjpfenninger**
Updates for Python 3.8 compatibility

Summary of changes in this pull request:

* Update dependencies for Python 3.8 on conda-forge
* Move pytest-xdist to base requirements and parallelise tests on Azure
* Requires changes to clustering example model tests due to a [fix in scikit-learn 0.22](https://scikit-learn.org/stable/whats_new/v0.22.html#sklearn-cluster) changing the behaviour of kmeans with a given random seed

Reviewer checklist:

- [ ] Test(s) added to cover contribution
- [ ] Documentation updated
- [ ] Changelog updated
- [ ] Coverage maintained or improved

### CODE REVIEW

File: changelog.rst [hunk 1]
```diff
@@ -40,6 +40,8 @@ Release History
 
 |fixed| Fix an issue that prevented 0.6.4 from loading NetCDF models saved with older versions of Calliope. It is still recommended to only load models with the same version of Calliope that they were saved with, as not all functionality will work when mixing versions.
 
+|fixed| Updated to require pandas 0.25, xarray 0.14, and scikit-learn 0.22, and verified Python 3.8 compatibility. Because of a bugfix in scikit-learn 0.22, models using k-means clustering with a specified random seed may return different clusters from Calliope 0.6.5 on.
```
Comment (brynpickering) [inline changelog.rst]: Should we flag this as possibly backwards-incompatible?
